adapta/storage/blob/s3_storage_client.py

```diff
@@ -132,15 +159,85 @@ def download_blobs(
         filter_predicate: Optional[Callable[[...], bool]] = None,
     ) -> None:
         """
-        Not yet implemented in S3 Client
+        Downloads blobs from S3 storage to a local path.
+
+        :param blob_path: Path to blob
+        :param local_path: Local path to download the blobs to
+        :param threads: Number of threads to use for the download
+        :param filter_predicate: Optional callable to filter blobs
+        :return:
+        """
+        s3_path = cast_path(blob_path)
+        blobs = self._s3_resource.Bucket(s3_path.bucket).objects.filter(Prefix=s3_path.path)
+        for blob in blobs:
+            if filter_predicate is None or filter_predicate(blob):
+                local_file_path = f"{local_path}/{blob.key}"
+                os.makedirs(os.path.dirname(local_file_path), exist_ok=True)
+                try:
+                    self._s3_resource.meta.client.download_file(s3_path.bucket, blob.key, f"{local_path}/{blob.key}")
+                except StorageClientError as error:
+                    raise RuntimeError(f"Error downloading blob: {error}") from error
+
+    def copy_blob(self, blob_path: DataPath, target_blob_path: DataPath, doze_period_ms: int = 0) -> None:
```

Comment:
Perhaps we should indicate in the docstring that doze_period_ms is not used in the AWS client and exists solely for interface compatibility (?)

Reply:
That is a very good idea 👍 I've updated the descriptions in [6714a99](https://github.com/SneaksAndData/adapta/pull/437/commits/6714a995ab99157e76d71dbeaa596e404b15d640)
Thanks for the input 🙂